### DISCUSSION TITLE
Support for "Water Leak Detector"

### DISCUSSION THREAD

**Discussion by Nightreaver** [arendst/Tasmota#13070] [Ideas]
I noticed that the sensor I bought recently support WIFI, so I have been checking the inside and I saw an ESP-12(?).

I measured the PINs and wanted to ask if support is possible?

![image](https://user-images.githubusercontent.com/3270156/132132559-b1f0deee-ab33-4b01-af70-1ad80b149c8b.png)
![pins_front](https://user-images.githubusercontent.com/3270156/132132385-d26bdbf8-005a-4494-b981-11e4b9c58fbe.png)
![pins_back_b](https://user-images.githubusercontent.com/3270156/132132615-46aefa28-5b56-4b3f-b770-45536e738897.png)

The LED turns on when water is detected, but also the buzzer is on during that time. The buzzer was one of the main reasons I decided for this model instead of the others.

Battery is an CR123A with 3V

Of course its available on aliexpress and probably as well on banggood and others.

https://www.aliexpress.com/item/1005002889533009.html

I can try to backup original firmware if that is helpful, and I also can try to flash tasmota if I can get some info on how to set it up.

**Comment by sfromis** [comment]
That [XR3](https://developer.tuya.com/en/docs/iot/xr3-datasheet?id=K98s9168qi49g) module is not ESP-12, and is not compatible with Tasmota. I'd not be surprised if it is pin-compatible with ESP-12, with the potential for relatively easy replacement. Note that battery-operated devices are not a good fit for ESP8266, and there is more than a minor risk of getting horrible battery lifetime if frankensteining it.
